### PR TITLE
Add decision record about JavaScript file names and formats

### PR DESCRIPTION
**Pull request by colinrotherham**
Closes: https://github.com/alphagov/govuk-design-system-architecture/issues/26

### CODE REVIEW

File: decision-records/007-javascript-file-formats.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+We will add our [GitHub release](https://github.com/alphagov/govuk-frontend/releases) JavaScript to the [`govuk-frontend` npm package](https://www.npmjs.com/package/govuk-frontend).
+
+**Date:** 2023-06-27
+
+**Status:** accepted
+
+## Rationale
+
+We can avoid future breaking changes by changing by file names and extensions to clearly show:
```
Comment (claireashworth) [inline decision-records/007-javascript-file-formats.md:30]:
```suggestion
We can avoid future breaking changes by changing file names and extensions to clearly show:
```

File: decision-records/007-javascript-file-formats.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+We will publish JavaScript in the following formats:
+
+1. ECMAScript (ES) modules
+2. ECMAScript (ES) modules, bundled
+3. Universal Module Definition (UMD), bundled
+
+We will use ECMAScript (ES) modules by default and only bundle JavaScript files that are considered "entry points" into GOV.UK Frontend, such as `all.mjs` and exported components (for example `accordion.mjs`).
+
+We will deprecate the Universal Module Definition (UMD) format in a future release.
```
Comment (36degrees) [inline decision-records/007-javascript-file-formats.md:15]: We talked about this, but I think we should focus decision records on things we've decided to do now, and avoid making future-looking statements.

(I'd be happy to mention it as a future possibility elsewhere in the doc, just not including it in the decision)

File: decision-records/007-javascript-file-formats.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+For consistency, we will clearly update our file extensions to:
+
+1. Add prefix `.bundle` to bundled JavaScript file extensions
+2. Add prefix `.min` to minified JavaScript file extensions
+
+We will add our [GitHub release](https://github.com/alphagov/govuk-frontend/releases) JavaScript to the [`govuk-frontend` npm package](https://www.npmjs.com/package/govuk-frontend).
```
Comment (36degrees) [inline decision-records/007-javascript-file-formats.md:22]: Think it's worth being clearer about what's being included from the package – as far as I recall it's just the minified JS?

File: decision-records/007-javascript-file-formats.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+**Date:** 2023-06-27
+
+**Status:** accepted
```
Comment (36degrees) [inline decision-records/007-javascript-file-formats.md:26]: I've raised #29 to fix this in the template, but can be bump the date and status to the top of the doc, and sentence case the status?

File: decision-records/007-javascript-file-formats.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
+# JavaScript file names and formats
+
+## Decision
```
Comment (36degrees) [inline decision-records/007-javascript-file-formats.md:3]: This is looking great so far. Do you think it'd be useful to include the same tree view of the structure of the package we have in https://github.com/alphagov/govuk-frontend/pull/3726 as well?

File: decision-records/007-javascript-file-formats.md (hunk 3 above)
Comment (colinrotherham) [reply decision-records/007-javascript-file-formats.md]: Sorry, our paths crossed and it was a post-sports day change to add the directory listing example 😊

Pushed up a bit more now

File: decision-records/007-javascript-file-formats.md (hunk 2 above)
Comment (colinrotherham) [reply decision-records/007-javascript-file-formats.md:21]: Ah this was one of @romaricpascal's decisions that we'd deprecate UMD + window globals in future

Will remove

File: decision-records/007-javascript-file-formats.md (hunk 4 above)
Comment (colinrotherham) [reply decision-records/007-javascript-file-formats.md]: Spotted it, pushed 👍

File: decision-records/007-javascript-file-formats.md (hunk 1 above)
Comment (colinrotherham) [reply decision-records/007-javascript-file-formats.md]: Thanks @claireashworth, still chipping away at this but fixed this one

**Comment by colinrotherham** [comment]
Bringing this back out of draft now @36degrees @claireashworth

**Review by 36degrees — APPROVED** [review]

**Comment by colinrotherham** [comment]
Thanks @36degrees

Do we need to mention that the bundled code is still full of (quite large) comment blocks?

**Comment by 36degrees** [comment]
I don't think so, unless you think it's relevant for the overall decision (i.e. if it was somehow a factor in what we decided, or is tied in the implications)?

**Comment by colinrotherham** [comment]
It was flagged during, but not really related to this decision

**Review by claireashworth — APPROVED** [review]

**Pull request merged**